### PR TITLE
WIP: One sided payments

### PR DESCRIPTION
**Pull request by CjS77**
Pushing progress now. 

I'll rebase into "bite-ish sized" sets of commits when ready for review.

### CODE REVIEW

File: applications/tari_app_grpc/src/blocks.rs [hunk 1]
```diff
@@ -63,10 +63,11 @@ impl From<tari_core::blocks::Block> for grpc::Block {
                     .iter()
                     .map(|input| grpc::TransactionInput {
                         features: Some(grpc::OutputFeatures {
-                            flags: input.features.flags.bits() as u32,
-                            maturity: input.features.maturity,
+                            flags: input.features().flags.bits() as u32,
+                            maturity: input.features().maturity,
                         }),
-                        commitment: Vec::from(input.commitment.as_bytes()),
+                        commitment: Vec::from(input.commitment().as_bytes()),
```
Comment (sdbondi) [inline applications/tari_app_grpc/src/blocks.rs:69]: nit: (In a few places)
```suggestion
                        commitment: input.commitment().to_vec(),
```

File: applications/tari_base_node/src/parser.rs [hunk 2]
```diff
@@ -554,17 +553,15 @@ impl Parser {
                             "\nYou have {} UTXOs: (value, commitment, mature in ? blocks, flags)",
                             unspent_outputs.len()
                         );
-                        let factory = PedersenCommitmentFactory::default();
+                        let _factory = PedersenCommitmentFactory::default();
```
Comment (sdbondi) [inline applications/tari_base_node/src/parser.rs]:
```suggestion

```

**Review by sdbondi — COMMENTED** [review]
Nice 🖇️  hunting - IMHO these changes should go in as a clean up PR and any further work should be done in a new PR.  Just need the broken tari_crypto cargo.toml references to be removed and the tests to pass.

**Comment by phyro** [comment]
Hi @CjS77 
I'm not sure this is the appropriate place, but your links (https://rfc.tari.com/) to RFC Pull Requests link to a 404. I just wanted to say that in [RFC-0201](https://rfc.tari.com/RFC-0201_TariScript.html), the approach to commit to additional data `One approach to commit to the script hashes is to modify the output commitments using the data commitments approach suggested by Phyro.` is very likely flawed. You can read more https://phyro.github.io/grinvestigation/data_commitments.html at the top of the page why that is. I'm not sure if you guys are using this method or not, but anything found on Grinvestigation page should be assumed insecure and would need to be reviewed by a cryptographer. I'm sorry if this warning is late, I wasn't aware this was referenced in an RFC until someone showed me. I hope this doesn't impact your work too much 🤞

**Comment by CjS77** [comment]
Thanks for the heads up @phyro. Appreciate you reaching out. This PR isn't merged, so it doesn't affect any live code.

Who spotted the attack, btw?

**Comment by phyro** [comment]
@CjS77 It was a member of the Grin community called Kurt - I don't think he's on Github. It's been found some time ago, but I never updated the page because I didn't know anyone was looking at this.

**Comment by CjS77** [comment]
Will re-open using updated script schema

**Pull request closed without merging**